website/content/docs/concepts/cpu.mdx

```diff
@@ -0,0 +1,317 @@
+```
+
+On Linux systems, Nomad supports reserving whole CPU cores specifically for a
+task. No task will be allowed to run on a CPU core reserved for another task.
+
+```hcl
+task {
+  resources {
+    cores = 4
+  }
+}
+```
+
+Nomad Enterprise supports NUMA aware scheduling, which enables operators to
+more finely control which CPU cores may be reserved for tasks.
+
+### CPU Hard Limits
+
+Some task drivers support the configuration option `cpu_hard_limit`. If enabled
+this option will restrict tasks from bursting above their CPU limit even
+when there is idle capacity on the node. The tradeoff is consistency versus
+utilization. A task with too few CPU resources may operate fine until another
+task is placed on the node causing a reduction in available CPU bandwidth, which
+could cause distruption for the underprovisioned task.
+
+### CPU Environment Variables
+
+To help tasks understand the resources available to them, Nomad sets environment
+variables in the runtime environment of each task.
+
+
+- `NOMAD_CPU_LIMIT` - The amount of CPU bandwidth allocated on behalf of the
+task.
+
+- `NOMAD_CPU_CORES` - The set of cores in [cpuset][] notation reserved for the task.
+This value is only set if `resources.cores` is configured.
+
+```env
+NOMAD_CPU_CORES=3-5
+NOMAD_CPU_LIMIT=9000
+```
+
+# NUMA
+
+Nomad clients are commonly provisioned on real hardware in an on-premise
+environment or in the cloud on large `.metal` instance types. In either case it
+is likely the underlying server is designed around a [NUMA topology][numa_wiki].
+Any server that contains multiple CPU sockets or multiple RAM banks per CPU
+socket will be characterized by the non-uniform access times involved in
+accessing system memory.
+
+
+[![NUMA](/img/nomad-numa.png)](/img/nomad-numa.png)
+
+The simplified example machine above has the following topology
+- 2 physical CPU sockets
+- 4 system memory banks, 2 per socket
+- 8 physical cpu cores (4 per socket)
+- 2 logical cpu cores per physical core
+- 4 PCI devices, 1 per memory bank
+
+### optimizing performance
+
+Any OS process scheduled on core X where X belongs to  memory node M will take
+longer to access any memory node that is not M.
+
+Using the example above if a task is scheduled on Core 0, accessing memory in
+Mem 1 might take 20% longer than accessing memory in Mem 0, and accessing memory
+in Mem 2 might take 300% longer. The extreme differences are due to various
+physical hardware limitations. A core accessing memory in its own NUMA node is
+optimal. Programs which perform a high throughput of reads or writes to/from
+system memory will have their performance substantially hindered by not optimizing
+their spatial locality iwth regard to the systems NUMA topology.
+
+### SLIT tables
+
+Any modern machine will define System Locality Distance Information (SLIT)
+tables in its firmware. These tables are understood and made referenceable
+by the Linux kernel. There are two key pieces of information provided by SLIT
```

Comment:
Boo...

> The relative distance between nodes on a system is stored in the ACPI System Locality Distance Information Table (SLIT), which is not exposed by any Windows functions.

https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getnumaproximitynode

I was curious if this was Linux-only.

Reply:
Well as long as we can associate core with NUMA node we can make `affinity = require` work on Windows. But without the SLIT data `affinity = prefer` will not work.

Reply:
Or at least, not well :sweat_smile: